client/post-editor/editor-more-options/copy-post.jsx

```diff
@@ -58,54 +55,65 @@ class EditorMoreOptionsCopyPost extends Component {
 	}
 
 	goToNewDraft = () => {
-		if ( '' !== this.state.selectedPostId ) {
-			page.redirect( `/post/${ this.props.siteSlug }?copy=${ this.state.selectedPostId }` );
+		const { siteSlug, type } = this.props;
+		const { selectedPostId } = this.state;
+		if ( '' !== selectedPostId ) {
+			page.redirect( `/${ type }/${ siteSlug }?copy=${ selectedPostId }` );
 			this.closeDialog();
 		}
 	}
 
 	render() {
-		const { translate, siteId } = this.props;
+		const { siteId, translate, type } = this.props;
+		const { selectedPostId, showDialog } = this.state;
 		const buttons = [ {
 			action: 'cancel',
 			label: translate( 'Cancel' ),
 		}, {
 			action: 'copy',
 			label: translate( 'Overwrite' ),
 			isPrimary: true,
-			disabled: ! this.state.selectedPostId,
+			disabled: ! selectedPostId,
 			onClick: this.goToNewDraft,
 		} ];
 		return (
 			<AccordionSection className="editor-more-options__copy-post">
 				<EditorDrawerLabel
-					labelText={ translate( 'Copy Post' ) }
-					helpText={ translate( "Pick a post and we'll copy the title, content, tags and categories. " ) }
+					labelText={ translate( 'Copy %s', { args: capitalize( type ), comment: '"Post" or "Page"' } ) }
+					helpText={ translate(
+						"Pick a %s and we'll copy the title, content, tags and categories.",
```

Comment:
I am not sure if adding the translated post type will work well in all languages from a grammar standpoint. For example in a post and a page may be a different gender and this would lead to different conjugations of the verb.

I remember that in core, after some discussion, all messages were duplicated for all post types.